src/taker/api.rs

```diff
@@ -290,6 +301,37 @@ impl Taker {
         })
     }
 
+    /// Helper functions to load and save the offer cache.
+    pub fn load_offer_cache(&self, path: &PathBuf) -> Result<Vec<CachedOffer>, IoError> {
+        if path.exists() {
+            let data = fs::read_to_string(path)?;
+            let offers: Vec<CachedOffer> = serde_json::from_str(&data).map_err(IoError::other)?;
+            Ok(offers)
+        } else {
+            Ok(vec![])
+        }
+    }
+
+    /// Saves the offer cache
+    pub fn save_offer_cache(
+        &self,
+        path: &PathBuf,
+        offers: &Vec<CachedOffer>,
+    ) -> Result<(), IoError> {
+        let data = serde_json::to_string_pretty(offers).map_err(IoError::other)?;
+        fs::write(path, data)
+    }
+
+    #[allow(dead_code)]
+    /// Deletes the offerbook cache file if it exists.
+    pub fn delete_offer_cache(&self) -> Result<(), IoError> {
+        let cache_file = get_taker_dir().join("offerbook_cache.json");
+        if cache_file.exists() {
+            fs::remove_file(cache_file)?;
+        }
+        Ok(())
+    }
+
```

Comment:
There is no need for these new APIs. The `OfferBook` structure already has read/write APIs. Just put the same logic in the old API. Otherwise, it will create redundant file io.